projects/components/src/table/table-api.ts

```diff
@@ -1,26 +1,32 @@
 import { Dictionary } from '@hypertrace/common';
 import { Observable } from 'rxjs';
 import { FilterAttribute } from '../filter-bar/filter-attribute';
-import { TableCellParserConstructor } from './cells/table-cell-parser';
+import { TableCellParserBase } from './cells/table-cell-parser-base';
 import { TableCellRendererConstructor } from './cells/table-cell-renderer';
 import { TableCellAlignmentType } from './cells/types/table-cell-alignment-type';
 
 export interface TableColumnConfig {
-  field: string;
+  field: string; // This is the unique name for the column (usually same as name except for composite fields)
+  name?: string; // Attribute name (for composite fields this should be the filterable attribute)
   display?: string;
-  renderer?: TableCellRendererConstructor;
-  parser?: TableCellParserConstructor<unknown, unknown, unknown>;
-  filterAttribute?: FilterAttribute;
   title?: string;
   titleTooltip?: string;
   sort?: TableSortDirection;
   visible?: boolean;
   sortable?: boolean;
+  filterable?: boolean;
   alignment?: TableCellAlignmentType;
   width?: number | string;
   onClick?(row: TableRow, column: TableColumnConfig): void;
 }
 
+export interface TableColumnConfigExtended extends TableColumnConfig {
```

Comment:
What happens if we don't export this? The idea of having an internal version of the column config is to avoid expoing implementation details, but this interface just exploded all over this commit and makes it even more confusing. At the least, anything that gets plugged into the table - i.e. parsers, renderers, columns etc. - should be working with a regular column, and anything that gets emitted should be a regular column.

Reply:
Fixed.